### PR TITLE
omnisharp-roslyn: use buildDotnetModules

### PR DESCRIPTION
**Pull request by mdarocha**
###### Description of changes

This refactors the omnisharp-roslyn package to use the `buildDotnetModules` pattern.

Since this package uses non-nuget package sources, I needed to update the `nuget-to-nix` and `make-nuget-deps` to support this.

This change also causes a small breaking change, since the name of the resulted binary changed from `omnisharp` to `OmniSharp`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mdarocha** [comment]
I'll also look into enabling tests during the check phase

File: pkgs/build-support/dotnet/build-dotnet-module/default.nix
Comment (mdarocha) [inline pkgs/build-support/dotnet/build-dotnet-module/default.nix:126]: I'm not really sure of this line. It's pretty hacky.

omnisharp-roslyn contains a postFixup hook which edits the global.json file - the default one upstream doesn't work with our dotnet version (it's pretty strict, I'm not sure why).

Since the `fetch-deps` passthru script uses the `src` attribute without patches, it doesnt work without removing the global.json file.

A better solution would be to use a patched source in this script, one that has all the patches and `preFixup`/`postFixup` hooks applied. I'm not sure how to do this hovewer.

Comment (SuperSandro2000) [reply pkgs/build-support/dotnet/build-dotnet-module/default.nix:126]:
```suggestion
      rm -f global.json
```

also please add a short comment

**Comment by SuperSandro2000** [comment]
Thanks for cleaning this up

**Review by ericdallo — APPROVED** [review]

File: pkgs/build-support/dotnet/build-dotnet-module/default.nix
Comment (mdarocha) [reply pkgs/build-support/dotnet/build-dotnet-module/default.nix]: I used `srcOnly` instead, this seems like a more generic fix.

**Review by IvarWithoutBones — COMMENTED** [review]
Thanks a lot for these changes, this is looking great. It's really nice to see some more packages being converted to `buildDotnetModule`. I also had no idea about `srcsOnly`, it definitely seems like a very clean solution for the custom nuget sources :blush:

After running nixpkgs-review on this I noticed `openvino` and its python modules fail to build, although I dont think those are regressions caused by this PR. If you could confirm thats the case and resolve the conflicts I'll gladly merge this :)

Also, could you squash the `omnisharp-dotnet: use buildDotnetModule` and `omnisharp-rosylin: add mdarocha to maintainers` commits? `build-dotnet-module` -> `buildDotnetModule` for https://github.com/NixOS/nixpkgs/pull/178004/commits/70b863f0ef933e702215c38926026c2c75c577f3 is also a bit more consistent with the previous commit descriptions.

**Comment by mdarocha** [comment]
> After running nixpkgs-review on this I noticed `openvino` and its python modules fail to build, although I dont think those are regressions caused by this PR. If you could confirm thats the case and resolve the conflicts I'll gladly merge this :)

`openvivo` uses `srcOnly`, which seems to have triggered the rebuild. But as I checked, this package also fails to build on latest `master`, and the `onnx_src` attribute seems to generate correctly. So I don't think this is caused by these changes.

> Also, could you squash the `omnisharp-dotnet: use buildDotnetModule` and `omnisharp-rosylin: add mdarocha to maintainers` commits? `build-dotnet-module` -> `buildDotnetModule` for [70b863f](https://github.com/NixOS/nixpkgs/commit/70b863f0ef933e702215c38926026c2c75c577f3) is also a bit more consistent with the previous commit descriptions.

Done

**Comment by IvarWithoutBones** [comment]
@ofborg eval

**Comment by mdarocha** [comment]
I rebased on top of master, and ofborg seems fixed now

**Review by IvarWithoutBones — APPROVED** [review]
LGTM, once again thank you for the work :+1: 

Result of `nixpkgs-review pr 178004` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>openvino</li>
    <li>python310Packages.openvino</li>
    <li>python39Packages.openvino</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>nimlsp</li>
    <li>nuget-to-nix</li>
    <li>omnisharp-roslyn</li>
  </ul>
</details>

**Pull request merged**
